File: Lab 1/task 1/task1.java
import java.util.Scanner;

public class task1 {
    public static int summation(int[] arr) {
        int sum = 0;
        for(int num : arr) {
            sum += num;
        }
        return sum;
    }

    public static int maximum(int[] arr) {
        int max = arr[0];
        for(int num : arr) {
            if(num > max) max = num;
        }
        return max;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter the number of elements in the array: ");
        int n = scanner.nextInt();
        int[] arr = new int[n];
        for(int i = 0; i < n; i++) {
            System.out.print("Enter element " + (i+1) + ": ");
            arr[i] = scanner.nextInt();
        }
        System.out.println("Summation: " + summation(arr));
        System.out.println("Maximum: " + maximum(arr));
    }
}

File: Lab 1/task 1/task1.cpp
#include <iostream>
#include <vector>
using namespace std;

int summation(const vector<int>* arr) {
    int sum = 0;
    for(int num : *arr) {
        sum += num;
    }
    return sum;
}

int maximum(const vector<int>* arr) {
    int max = (*arr)[0];
    for(int num : *arr) {
        if(num > max) max = num;
    }
    return max;
}

int main() {
    int n;
    cout << "Enter the number of elements in the array: ";
    cin >> n;
    vector<int> arr(n);
    for(int i = 0; i < n; ++i) {
        cout << "Enter element " << i+1 << ": ";
        cin >> arr[i];
    }
    cout << "Summation: " << summation(&arr) << endl;
    cout << "Maximum: " << maximum(&arr) << endl;
    return 0;
}
